amsclient/amsclient.go

```diff
@@ -73,12 +73,18 @@ type AMSClient interface {
 
 // amsClientImpl is an implementation of the AMSClient interface
 type amsClientImpl struct {
-	connection *sdk.Connection
-	pageSize   int
+	connection         *sdk.Connection
+	pageSize           int
+	clusterListCaching bool
 }
 
 // NewAMSClient create an AMSClient from the configuration
 func NewAMSClient(conf Configuration) (AMSClient, error) {
+	if conf.ClusterListCaching {
```

Comment:
You can use `log.Info().Bool("Enabled", conf.ClusterListCaching).Msg("Caching for cluster list")` for less lines of code :)

Reply:
Thanks for advice I changed it :D